### PR TITLE
add code tabs in num20.

### PR DESCRIPTION
**Pull request by benluo**

### CODE REVIEW

File: _overviews/scala3-book/domain-modeling-tools.md [hunk 1]
```diff
@@ -21,56 +22,108 @@ Scala 3 provides many different constructs so we can model the world around us:
 
 This section briefly introduces each of these language features.
 
-
 ## Classes
 
 As with other languages, a _class_ in Scala is a template for the creation of object instances.
 Here are some examples of classes:
 
+{% tabs class_1 class=tabs-scala-version %}
```
Comment (bishabosha) [inline _overviews/scala3-book/domain-modeling-tools.md:30]: `class=tabs-scala-version` should not be used for "Scala 2 and 3" tabs. Soon I am adding assertions to make CI fail for this
```suggestion
{% tabs class_1 %}
```

File: _overviews/scala3-book/domain-modeling-tools.md [hunk 2]
```diff
@@ -21,56 +22,108 @@ Scala 3 provides many different constructs so we can model the world around us:
...
 //      ---
 ```
 
+{% endtab %}
+{% endtabs %}
+
 However, with [creator applications][creator] this isn’t required in Scala 3:
 
+{% tabs class_3 %}
+{% tab 'Scala 2 and 3' for=class_3 %}
```
Comment (bishabosha) [inline _overviews/scala3-book/domain-modeling-tools.md:63]:
```suggestion
{% tab 'Scala 3 Only' for=class_3 %}
```

File: _overviews/scala3-book/domain-modeling-tools.md [hunk 3]
```diff
@@ -196,17 +325,21 @@ The class has three constructors, given by the numbered comments in the code:
 
 Those constructors can be called like this:
 
+{% tabs structor_2 %}
+{% tab 'Scala 2 and 3' for=structor_2 %}
+
 ```scala
 val s1 = Student("Mary", "123")
```
Comment (bishabosha) [inline _overviews/scala3-book/domain-modeling-tools.md:332]: scala 2 needs to use `new`

Comment (benluo) [reply _overviews/scala3-book/domain-modeling-tools.md:342]: changed.

**Review by bishabosha — APPROVED** [review]
Thanks! I think a follow up PR can deal with the enums to provide scala 2 alternative

**Review by bishabosha — APPROVED** [review]
Thanks, I think a separate PR can handle the lack of enums in scala 2

**Pull request merged**